crates/router/src/connector/utils.rs

```diff
@@ -352,6 +370,102 @@ pub fn get_header_key_value<'a>(
         ))?
 }
 
+pub fn to_boolean(string: String) -> bool {
+    let str = string.as_str();
+    match str {
+        "true" => true,
+        "false" => false,
+        "yes" => true,
+        "no" => false,
+        _ => false,
+    }
+}
+
+pub fn get_connector_meta(
+    connector_meta: Option<serde_json::Value>,
+) -> Result<serde_json::Value, Error> {
+    connector_meta.ok_or_else(missing_field_err("connector_meta_data"))
+}
+
+pub fn to_connector_meta<T>(connector_meta: Option<serde_json::Value>) -> Result<T, Error>
+where
+    T: serde::de::DeserializeOwned,
+{
+    let json = connector_meta.ok_or_else(missing_field_err("connector_meta_data"))?;
+    parse_struct(json)
+}
+
+pub fn to_connector_meta_from_secret<T>(
+    connector_meta: Option<Secret<serde_json::Value>>,
+) -> Result<T, Error>
+where
+    T: serde::de::DeserializeOwned,
+{
+    let connector_meta_secret =
+        connector_meta.ok_or_else(missing_field_err("connector_meta_data"))?;
+    let json = connector_meta_secret.peek().clone();
+    parse_struct(json)
+}
+
+pub fn parse_struct<T>(json: serde_json::Value) -> Result<T, Error>
+where
+    T: serde::de::DeserializeOwned,
+{
+    serde_json::from_value::<T>(json.clone())
+        .into_report()
+        .change_context(errors::ConnectorError::ParsingFailed {
+            from_type: "Json",
+            to_type: std::any::type_name::<T>(),
+            data: json.to_string(),
+        })
+}
+
+pub fn parse_struct_from_bytes<T>(bytes: bytes::Bytes) -> Result<T, Error>
+where
+    T: serde::de::DeserializeOwned,
+{
+    serde_json::from_slice::<T>(bytes.chunk())
+        .into_report()
+        .change_context(errors::ConnectorError::ParsingFailed {
+            from_type: "Bytes",
+            to_type: std::any::type_name::<T>(),
+            data: String::from_utf8(bytes.to_vec())
+                .into_report()
+                .change_context(errors::ConnectorError::ResponseHandlingFailed)?,
+        })
+}
+
+pub fn parse_struct_from_bytes_slice<T>(bytes: &[u8]) -> Result<T, Error>
+where
+    T: serde::de::DeserializeOwned,
+{
+    serde_json::from_slice::<T>(bytes)
```

Comment:
there is crate which has common utils function, check this file ` crates/common_utils/src/ext_traits.rs`

Reply:
we need this to handle connector error, as we duplication connector error handling in all the places it will solve the problem